core/templates/services/exploration-html-formatter.service.ts

```diff
@@ -126,7 +126,7 @@ export class ExplorationHtmlFormatterService {
   }
 
   getAnswerHtml(
-      answer: string, interactionId: string,
+      answer: string, interactionId: string | null,
```

Comment:
Why can `interactionId` be `null`?

Reply:
I looked at here (https://github.com/oppia/oppia/blob/develop/core/templates/domain/exploration/InteractionObjectFactory.ts#L95) and saw that id is defined to be string | null. Therefore, I added it here as well, and without adding it, running typescript checks fail since we are calling getAnswerHtml function in the input-respone-pair.component.ts file.

Reply:
Hmm, I agree, but on the other hand, I think we should modify the _InteractionObjectFactory.ts _ to only accept string ID, can you look into it and if it is not easy create an issue and TODO comment for it?

Reply:
Also, no matter the refactor lets make sure that in all places where we call `getAnswerHtml` it is called with string.

Reply:
So, I checked, and modifying the _InteractionObjectFactory.ts _ to only accept string ID will involve lot more changes, so I will create an issue. About the follow up thought, I checked and getAnswerHtml as a function, whereever it's being called from, we are indeed passing a string only, and not a null value for the interactionId. Did you have something else in mind?

Reply:
Can you then change the type to just `string`? If it causes some TS issues then solve them in the place where we call `getAnswerHtml`

Reply:
As discussed, I added the check **not** in the function where we call it but where it's defined (i.e. explorationHtmlFormatter.getAnswerHtml).